### ISSUE TITLE
Can't parse React composition using { nestedElement } construct inside JSX element

### ISSUE THREAD

**Issue by mbchoa** [team-gryff/react-monocle#38]
Monocle is unable to parse components which use dynamic composition to render nested child components.

For example:

```javascript
render: function () {
  const items = this.props.elements.map(element => <ListItem/>);
  return <div>
    {items}
  </div>
}
```

**Comment by jerrymao15** [comment]
This works now. However, we are still unable to determine how many times a component should be rendered (ie if you're rendering multiple same components).

**Comment by jerrymao15** [comment]
addressed with #77

**Comment by Bertg** [comment]
@jerrymao15 

A bit late to the party. But I was wondering about something related.

Because of the static nature of this project, runtime decisions are difficult of not impossible to detect. We for instance have several components who's job it is to select the correct component to render based on some property. I don't know if this would be possible to do. 

Would it make sense to be able to apply the visualisation on a live running react app? The react-dev-tools show a live representation of the react structure, so it should be possible.

The monocle visualisation could then be rendered in a floating full page div or something like that. I think that would be super useful.
